allennlp/data/dataset_readers/text_classification_json.py

```diff
@@ -0,0 +1,121 @@
+        like the Hierarchical Attention Network.
+    sequence_length: ``int``, optional (default = ``None``)
+        If specified, will truncate tokens to specified maximum length.
+    skip_label_indexing: ``bool``, optional (default = ``False``)
+        Whether or not to skip label indexing. You might want to skip label indexing if your
+        labels are numbers, so the dataset reader doesn't re-number them starting from 0.
+    lazy : ``bool``, optional, (default = ``False``)
+        Whether or not instances can be read lazily.
+    """
+    def __init__(self,
+                 token_indexers: Dict[str, TokenIndexer] = None,
+                 tokenizer: Tokenizer = None,
+                 segment_sentences: bool = False,
+                 sequence_length: int = None,
+                 skip_label_indexing: bool = False,
+                 lazy: bool = False) -> None:
+        super().__init__(lazy=lazy)
+        self._tokenizer = tokenizer or WordTokenizer()
+        self._segment_sentences = segment_sentences
+        self._sequence_length = sequence_length
+        self._skip_label_indexing = skip_label_indexing
+        self._token_indexers = token_indexers or {'tokens': SingleIdTokenIndexer()}
+        if self._segment_sentences:
+            self._sentence_segmenter = SpacySentenceSplitter()
+
+    @overrides
+    def _read(self, file_path):
+        with open(cached_path(file_path), "r") as data_file:
+            for line in data_file.readlines():
+                if not line:
+                    continue
+                items = json.loads(line)
+                text = items["text"]
+                label = str(items["label"])
+                instance = self.text_to_instance(text=text, label=label)
+                if instance is not None:
+                    yield instance
+
+    def _truncate(self, tokens):
+        """
+        truncate a set of tokens using the provided sequence length
+        """
+        if len(tokens) > self._sequence_length:
+            tokens = tokens[:self._sequence_length]
+        return tokens
+
+    @overrides
+    def text_to_instance(self, text: str, label: str = None) -> Instance:  # type: ignore
+        """
+
+        Parameters
+        ----------
+        text : ``str``, required.
+            The text to classify
+        label ``str``, optional, (default = None).
+            The label for this text.
+
+        Returns
+        -------
+        An ``Instance`` containing the following fields:
+            tokens : ``TextField``
+                The tokens in the sentence or phrase.
+            label : ``LabelField``
+                The label label of the sentence or phrase.
+        """
+        # pylint: disable=arguments-differ
+        fields: Dict[str, Field] = {}
+        if self._segment_sentences:
+            sentences: List[Field] = []
+            sentence_tokens = self._sentence_segmenter.split_sentences(text)
```

Comment:
nit: not sure if `sentence_tokens` is the right name here

Reply:
`sentence_splits`?  `sentence_segmentations`?